packages/block-editor/src/store/selectors.js

```diff
@@ -3130,6 +3145,16 @@ export const getBlockEditingMode = createRegistrySelector(
 					select( blocksStore )
 				);
 				const isContent = hasContentRoleAttribute( name );
+
+				// All controlling blocks are treated as content only
+				// by default.
+				if (
+					! isContent &&
+					areInnerBlocksControlled( state, clientId )
+				) {
+					return 'contentOnly';
+				}
```

Comment:
There's no explanation as to how this links to `templateLock`, so I find it hard to understand why this check is located here. If controlled blocks are always contentOnly then maybe there could be one check right at the start of the function that returns early.

Reply:
Can't do at the start - the idea is to have the list view behavior on blocks with controlled inner blocks, when they are children of content only locked parents. Sure, I can add comments to explain. We don't want blocks with controlled inner blocks to be content only locked _all the time_ but only when they're inside a container which itself is a content only locking parent.

Reply:
> only when they're inside a container which itself is a content only locking parent.

'content only locking parent' can unfortunately mean quite a few things, so it's hard for me to review the code without fully understanding the context.